modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -0,0 +1,725 @@
+                operation_id,
+                LightningCommonInit::KIND.as_str(),
+                |_, _| LightningOperationMeta,
+                transaction,
+            )
+            .await?;
+
+        Ok(operation_id)
+    }
+
+    pub async fn subscribe_send(
+        &self,
+        operation_id: OperationId,
+    ) -> anyhow::Result<UpdateStreamOrOutcome<SendState>> {
+        let operation = self.client_ctx.get_operation(operation_id).await?;
+        let mut stream = self.notifier.subscribe(operation_id).await;
+        let client_ctx = self.client_ctx.clone();
+        let module_api = self.module_api.clone();
+
+        Ok(operation.outcome_or_updates(&self.client_ctx.global_db(), operation_id, move || {
+            stream! {
+                loop {
+                    if let Some(LightningClientStateMachines::Send(state)) = stream.next().await {
+                        match state.state {
+                            SendSMState::Funding(..) => yield SendState::Funding,
+                            SendSMState::Funded(..) => yield SendState::Funded,
+                            SendSMState::Success(preimage) => {
+                                // the preimage has been verified by the state machine previously
+                                assert!(state.contract.verify_preimage(&preimage));
+
+                                yield SendState::Success(preimage);
+                                return;
+                            },
+                            SendSMState::Refunding(out_points) => {
+                                yield SendState::Refunding;
+
+                                match client_ctx.await_primary_module_outputs(operation_id, out_points.clone()).await {
+                                    Ok(..) => {
+                                        yield SendState::Refunded;
+                                        return;
+                                    },
+                                    Err(..) => {
+                                        // The gateway may have incorrectly claimed the outgoing contract thereby causing
+                                        // our refund transaction to be rejected. Therefore, we check one last time if 
+                                        // the preimage is available before we enter the failure state.
+                                        if let Some(preimage) = module_api.await_preimage(
+                                            &state.contract.contract_id(),
+                                            0
+                                        ).await {
+                                            if state.contract.verify_preimage(&preimage) {
+                                                yield SendState::Success(preimage);
+                                                return;
+                                            }
+                                        }
+
+                                        yield SendState::Failure;
```

Comment:
We should probably include an error message on this too. This should never happen right?